benchmarks/benchmarks.rb

```diff
@@ -203,42 +223,50 @@ def positional_splat_keyword_and_double_splat_args(a, *args, b:, **kwargs)
-  benchmark_json.sort_by! { _1["name"] }
-
-  # Print headers based on the first benchmark_json
-  if i.zero?
-    benchmark_headers = benchmark_json.map do |benchmark_gem|
-      # Gem name is of the form:
-      # "MemoWise (1.1.0): ()"
-      # We use this mapping to get a header of the form
-      # "`MemoWise` (1.1.0)
-      gem_name_parts = benchmark_gem["name"].split
-      "`#{gem_name_parts[0]}` #{gem_name_parts[1][...-1]}"
+end
+
+[true, false].each do |github_comparison|
+  benchmark_jsons.each_with_index do |benchmark_json, i|
+    # We print a comparison table after we run each benchmark to copy into our
+    # README.md
+
+    # MemoWise will not appear in the comparison table, but we will use it to
+    # compare against other gems' benchmarks
+    memo_wise = benchmark_json.find { _1["name"].include?("MemoWise") }
+    benchmark_json -= [memo_wise]
+
+    github_main = benchmark_json.find { _1["name"].include?(GITHUB_MAIN) }
+    benchmark_json = github_comparison ? [github_main] : benchmark_json - [github_main]
+
+    # Sort benchmarks by gem name to alphabetize our final output table.
+    benchmark_json.sort_by! { _1["name"] }
+
+    # Print headers based on the first benchmark_json
+    if i.zero?
+      benchmark_headers = benchmark_json.map do |benchmark_gem|
+        # Gem name is of the form:
+        # "MemoWise (1.1.0): ()"
+        # We use this mapping to get a header of the form
+        # "`MemoWise` (1.1.0)
+        gem_name_parts = benchmark_gem["name"].split
+        "`#{gem_name_parts[0]}` #{gem_name_parts[1][...-1]}"
+      end.join("|")
+      puts "|Method arguments|#{benchmark_headers}|"
+      puts "#{'|--' * (benchmark_json.size + 1)}|"
+    end
+
+    output_str = benchmark_json.map do |bgem|
+      # "%.2f" % 12.345 => "12.34" (instead of "12.35")
+      #   See: https://bugs.ruby-lang.org/issues/12548
+      # 1.00.round(2).to_s => "1.0" (instead of "1.00")
+      #
+      # So to round and format correctly, we first use Float#round and then %
+      "%.#{N_RESULT_DECIMAL_DIGITS}fx" %
+        (memo_wise["central_tendency"] / bgem["central_tendency"]).round(N_RESULT_DECIMAL_DIGITS)
     end.join("|")
-    puts "|Method arguments|#{benchmark_headers}|"
-    puts "#{'|--' * (benchmark_json.size + 1)}|"
-  end
 
-  output_str = benchmark_json.map do |bgem|
-    # "%.2f" % 12.345 => "12.34" (instead of "12.35")
-    #   See: https://bugs.ruby-lang.org/issues/12548
-    # 1.00.round(2).to_s => "1.0" (instead of "1.00")
-    #
-    # So to round and format correctly, we first use Float#round and then %
-    "%.#{N_RESULT_DECIMAL_DIGITS}fx" %
-      (memo_wise["central_tendency"] / bgem["central_tendency"]).round(N_RESULT_DECIMAL_DIGITS)
-  end.join("|")
-
-  name = memo_wise["name"].partition(": ").last
-  puts "|`#{name}`#{' (none)' if name == '()'}|#{output_str}|"
+    name = memo_wise["name"].partition(": ").last
+    puts "|`#{name}`#{' (none)' if name == '()'}|#{output_str}|"
+  end
+  puts ""
```

Comment:
Is this last "puts" on purpose?

Reply:
Yep, it allows for space between the two outputs. Definitely let me know if you can think of a clearer way to do this

Reply:
Ah ok, maybe just a comment making the intent clear - I'll leave a suggestion